apps/www/registry/default/ui/textarea.tsx

```diff
@@ -1,24 +1,44 @@
+'use client'
 import * as React from "react"
+import { useEffect, useRef } from "react"
+import { mergeRefs } from "react-merge-refs"
 
 import { cn } from "@/lib/utils"
 
 export interface TextareaProps
-  extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {}
+  extends React.TextareaHTMLAttributes<HTMLTextAreaElement> { }
 
 const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
   ({ className, ...props }, ref) => {
+    const texteAreaRef = useRef<HTMLTextAreaElement>(null)
+
```

Comment:
Could use `React.useImperativeHandle` to merge the refs so you don't need to pull in an extra dependency.


```tsx
    const texteAreaRef = useRef<HTMLTextAreaElement>(null)
    React.useImperativeHandle(ref, () => texteAreaRef.current!);
```

```tsx
<textarea
        className={cn(
          "flex min-h-[80px] w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50",
          className
        )}
        ref={texteAreaRef}
        {...props}
      />
```

https://react.dev/reference/react/useImperativeHandle

Reply:
I know it, I just prefer other way due to readability. -1 dependency, this is also good.

Reply:
how can i make a dynamic textarea with react-hook-form, it handles ref by itself . how can i make it use my ref